### PR TITLE
Stop docker daemon, when running cri-o

### LINKED ISSUE #3068: FEATURE: Remove docker daemon from the buildroot iso
**Minikube version**: `minikube version: v0.28.2`

- **OS** :
```
Linux frezbo 4.17.12-200.fc28.x86_64 #1 SMP Fri Aug 3 15:01:13 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```

Bootstrapped a cluster using this command:
```
minikube -v 5 start --network-plugin=cni --container-runtime=cri-o --bootstrapper=kubeadm --vm-driver=kvm2
```

After `minikube ssh` , I see docker daemon running even though **CRIO-O** is used as runtime

### PR DESCRIPTION
**Pull request by afbjorklund**
Stop the docker daemon, when running an alternative container runtime (and vice-versa)

Remove the output from docker-env, since there is nothing responding on port 2376 anyway
`Cannot connect to the Docker daemon at tcp://*:2376. Is the docker daemon running?`

But as long as we are using docker machine, it will always be _started_ (by the provisioner)

Closes #3068

### CODE REVIEW

**Comment by afbjorklund** [comment]
Note that cri-o will fail to `kubeadm init` after this change, unless also merging: #3154

`Error starting cluster:  kubeadm init error`

File: cmd/minikube/cmd/start.go [hunk 1]
```diff
@@ -290,6 +290,31 @@ func runStart(cmd *cobra.Command, args []string) {
 		cmdutil.MaybeReportErrorAndExit(err)
 	}
 
+	fmt.Println("Stopping extra container runtimes...")
+
+	containerRuntime := viper.GetString(containerRuntime)
+	if config.VMDriver != "none" && containerRuntime != "" {
```
Comment (balopat) [inline cmd/minikube/cmd/start.go]: nit: can we combine the `config.VMDriver != "none" ifs into one?

File: cmd/minikube/cmd/start.go [hunk 2]
```diff
@@ -290,6 +290,31 @@ func runStart(cmd *cobra.Command, args []string) {
...
+		}
+	}
+	if config.VMDriver != "none" && (containerRuntime != "crio" && containerRuntime != "cri-o") {
+		if _, err := host.RunSSHCommand("sudo systemctl stop crio"); err != nil {
+			glog.Errorf("Error stopping crio", err)
+		}
+	}
+	if config.VMDriver != "none" && containerRuntime != "rkt" {
+		if _, err := host.RunSSHCommand("sudo systemctl stop rkt-api"); err == nil {
+			_, err = host.RunSSHCommand("sudo systemctl stop rkt-metadata")
```
Comment (balopat) [inline cmd/minikube/cmd/start.go]:
```suggestion
			_, err = host.RunSSHCommand("sudo systemctl stop rkt-api")
```

Comment (balopat) [reply cmd/minikube/cmd/start.go]: ahh ignore this comment, I misread it.

**Pull request merged**